### PR TITLE
fixed dialog race condition

### PR DESCRIPTION
**Pull request by lfdversluis**

### CODE REVIEW

**Comment by lfdversluis** [comment]
@whirm I think something is wrong with bbq. Apparently one of the failing tests is `Tribler.Test.test_gui_general.TestGuiGeneral.test_debugpanel` which by no means I can get it to crash. Is there an issue with this merger since I am PRing against next instead of devel? Because this PR is tester is apparently testing against devel?

**Comment by whirm** [comment]
The pull request tester job tests the PR, not a specific branch, I removed `_devel` from the name so people don't get mislead :)

**Comment by whirm** [comment]
I would start by fixing the "Session is singleton" error most tests fail with.

**Comment by lfdversluis** [comment]
:fireworks:

File: .gitignore [hunk 1]
```diff
@@ -29,3 +29,6 @@ flycheck_*
 memorydump.out
 test_file
 test_file.torrent
+
+# pycharm
+**.idea/
```
Comment (devos50) [inline .gitignore]: When will this one finally be merged into devel/next? :P

Comment (lfdversluis) [inline .gitignore]: about time, don't you think? :p Has been in 4 PRs so far lol

Comment (lfdversluis) [inline .gitignore]: Should've made a separate PR just for it :laughing:

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
@whirm ready. I left 2 commits because I did two things :) (actually 3 with the events but w/e)

File: Tribler/Core/Upgrade/upgrade.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import logging
 import os
 import shutil
-
```
Comment (whirm) [inline Tribler/Core/Upgrade/upgrade.py]: don't remove this

File: Tribler/Core/Upgrade/upgrade.py [hunk 3]
```diff
@@ -25,18 +25,52 @@ def __init__(self, session, db):
         self.session = session
         self.db = db
 
-        self.current_status = u"Checking Tribler version..."
         self.is_done = False
         self.failed = True
 
+        self.current_status = u"Checking Tribler version..."
+        if self.session.get_current_startup_config_copy().get_upgrader_enabled():
```
Comment (whirm) [inline Tribler/Core/Upgrade/upgrade.py]: Why are you making a copy of the startup config?

Comment (lfdversluis) [inline Tribler/Core/Upgrade/upgrade.py]: I believe calling `sessconfig` gives an exception.

File: Tribler/Core/Upgrade/upgrade.py [hunk 4]
```diff
@@ -49,36 +83,48 @@ def check_and_upgrade(self):
...
+            del torrent_store
+
+            self.failed = False
+            self.is_done = True
+            self.notify_done()
+        except Exception as e:
+            self._logger.exception(u"failed to upgrade: %s", e)
+
+    @call_on_reactor_thread
+    def stash_database(self):
```
Comment (whirm) [inline Tribler/Core/Upgrade/upgrade.py]: Just merge this and `_stash_database_away()` as it's only used here

**Comment by lfdversluis** [comment]
@whirm done.

**Comment by lfdversluis** [comment]
Yup I get a `AttributeError: CallbackConfigParser instance has no attribute 'get_upgrader_enabled'` now when using `sessconfig`. Reverting now

**Comment by whirm** [comment]
@lfdversluis also, you added a bunch of newlines all over the place, don't do that if not actually fixing code style guidelines as it gets annoying for other people when rebasing.

**Comment by lfdversluis** [comment]
@whirm Ok, ready I guess

**Pull request merged**